src/vs/workbench/common/resources.ts

```diff
@@ -224,6 +227,6 @@ export class ResourceGlobMatcher extends Disposable {
 			resourcePathToMatch = resource.fsPath; // TODO@isidor: support non-file URIs
 		}
 
-		return !!expressionForRoot && typeof resourcePathToMatch === 'string' && !!expressionForRoot(resourcePathToMatch);
+		return !!expressionForRoot && typeof resourcePathToMatch === 'string' && !!expressionForRoot(resourcePathToMatch, undefined, hasSibling);
```

Comment:
`!!expressionForRoot(resourcePathToMatch, undefined, hasSibling)` will always be true when the result is a promise